src/library_pthread.js

```diff
@@ -262,6 +262,42 @@ var LibraryPThread = {
         var worker = new Worker(pthreadMainJs);
 
         worker.onmessage = function(e) {
+          // TODO: Move the proxied call mechanism into a queue inside heap.
+          if (e.data.proxiedCall) {
+            var returnValue;
+            var funcTable = (e.data.func >= 0) ? proxiedFunctionTable : ASM_CONSTS;
+            var funcIdx = (e.data.func >= 0) ? e.data.func : (-1 - e.data.func);
+            PThread.currentProxiedOperationCallerThread = worker.pthread.threadInfoStruct; // Sometimes we need to backproxy events to the calling thread (e.g. HTML5 DOM events handlers such as emscripten_set_mousemove_callback()), so keep track in a globally accessible variable about the thread that initiated the proxying.
```

Comment:
Here I have an exception : `worker.pthread is undefined`, when a proxy call is lanch but the worker terminate and is clean.

Reply:
Any chance you'd be able to make a small test case?

Reply:
I test to reproduce it in a small test case but for now, i'm not able. I will continue to try.